flink-core/src/main/java/org/apache/flink/api/common/ArchivedExecutionConfig.java

```diff
@@ -54,6 +54,19 @@ public ArchivedExecutionConfig(ExecutionConfig ec) {
 		}
 	}
 
+	public ArchivedExecutionConfig(
+			String executionMode,
+			String restartStrategyDescription,
+			int parallelism,
+			boolean objectReuseEnabled,
+			Map<String, String> globalJobParameters) {
+		this.executionMode = executionMode;
```

Comment:
But isn't the other constructor already exactly trying to ensure what you say "simply won't happen"?

```java
if (ec.getGlobalJobParameters() != null
        && ec.getGlobalJobParameters().toMap() != null) {
    globalJobParameters = ec.getGlobalJobParameters().toMap();
} else {
    globalJobParameters = Collections.emptyMap();
}
```